interview-questions.md

```diff
@@ -8,42 +8,88 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
+Researched answer: a softeware development model that structures design around objects as opposed to functions and logic. An object is data that has unique behavior and its own attributes. 
 
-1. What is the difference between a Float and an Integer in Ruby?
+answers via: https://www.techtarget.com/searchapparchitecture/definition/object-oriented-programming-OOP#:~:text=Object%2Doriented%20programming%20(OOP)%20is%20a%20computer%20programming%20model,has%20unique%20attributes%20and%20behavior.
 
-Your answer:
 
-Researched answer:
+2. What is the difference between a Float and an Integer in Ruby?
 
-1. Ruby has an implicit return. What does that mean?
+Your answer: In Ruby there are two ways on desplaying a number. Float is the keyword that allows users to utilize decimal points. Integar is the keyword that refers to whole numbers. It will not give decimals and is the same as the number keyword, but integer is for the language Ruby. 
 
-Your answer:
+Researched answer: In ruby float is a number that contains the use of decimals. Any number without a decimal is a integer in Ruby. For example the number 24 is a integer, however the number 24.0 is a float. 
 
-Researched answer:
+answers via: https://www.digitalocean.com/community/tutorials/understanding-data-types-in-ruby
 
-1. What is a block in Ruby?
 
-Your answer:
+3. Ruby has an implicit return. What does that mean?
 
-Researched answer:
+Your answer: The language Ruby is written in a way that is simple to read. Implicit return is the languages way of returning the last line of code within a method without using any keywords. 
 
-1. How do you extract a value in a Ruby hash?
+Researched answer: Ruby will return the last line of a methods code. This is called the implicit return. In the example below we have a method called additon that has one parameter. inside the method we have a new variable called total which is holding num + num. On the next line we have 'total' which will automaticaly be the return because of Ruby implicit return. 
 
-Your answer:
+    def addition (num)
+        let total = num + num 
+        total
+    end
 
-Researched answer:
+answer via: https://www.learnhowtoprogram.com/ruby-and-rails/basic-ruby/defining-methods-in-ruby#:~:text=In%20Ruby%2C%20we%20can%20take,line%20before%20the%20end%20keyword.
+ 
+
+4. What is a block in Ruby?
+
+Your answer: A Ruby block is the flow of code from a parent/super method to its children. The super will have generic values that will be applicable to all children under that super.
+
+Researched answer: In Ruby blocks are bits of code inside of curly braces that are anonymous and can take input from arguments and return a value. 
+ex of structure: They can be executed by using the yeild key-word.  
+
+def yeild_block
+
+    puts 'Method'
+    yeild
+
+    put 'Method'
+    yeild
+end
+yeild_block{
+
+}
+
```

Comment:
A block is a function that gets passed to a method. You can think about them like higher-order functions in JS. For example, the `.map` action in Ruby takes a block (scoped between do / end) that passes a variable that will represent each item in the array during an iteration. The block is the function that is passed to the map method.